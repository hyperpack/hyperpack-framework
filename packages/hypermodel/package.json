{
  "name": "@hyperpack/hypermodel",
  "version": "1.0.0",
  "main": "src/index.ts",
  "license": "MIT",
  "scripts": {},
  "dependencies": {
    "class-transformer": "^0.2.3",
    "class-validator": "^0.14.0",
    "typeorm": "^0.2.24"
  },
  "devDependencies": {}
}