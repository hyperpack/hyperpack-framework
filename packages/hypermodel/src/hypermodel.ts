import { classToPlain } from 'class-transformer'
import { validate } from 'class-validator'
import { BaseEntity, BeforeInsert, BeforeUpdate } from 'typeorm'
import { ValidationException } from './validation-exception'

class HyperModel extends BaseEntity {
  @BeforeInsert()
  @BeforeUpdate()
  async beforeSave() {
    const err = await validate(this)
    if (err && err.length > 0) {
      throw new ValidationException(err)
    }
  }

  updateAttributes(attrs) {
    Object.assign(this, attrs)
  }

  asJson() {
    return classToPlain(this)
  }
}

export { HyperModel }
