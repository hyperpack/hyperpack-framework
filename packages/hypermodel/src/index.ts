export { HyperModel } from './hypermodel'
export { ValidationException } from './validation-exception'
export { createConnection } from './boot'
export { seed } from './seed'
export * from 'class-transformer'
export * from 'class-validator'
