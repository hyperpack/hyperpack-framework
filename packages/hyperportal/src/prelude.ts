import Purdy from 'purdy'
import { table } from 'table'
import L from 'lodash'
import asTable from 'as-table'

const purdy = Purdy.stringify

const objectify = m => x => {
  // just always use toJSON for sequelize.
  const invoke = o => (o[m] || o.asJson).bind(o)()

  if (L.isArray(x)) {
    return L.map(x, invoke)
  }
  return invoke(x)
}

const tablify = x => {
  const a = L.isArray(x) ? x : [x]
  const header = Object.keys(L.maxBy(a, _ => Object.keys(_).length))
  const rows = L.map(a, _ => L.map(header, p => _[p]))
  return [header, ...rows]
}

const prelude = log => {
  const o = _ =>
    _.then(objectify('toObject'))
      .then(purdy)
      .then(log)
  const js = _ =>
    _.then(objectify('asJson'))
      .then(purdy)
      .then(log)
  const t = _ =>
    _.then(objectify('toObject'))
      .then(tablify)
      .then(table)
      .then(log)
  const t1 = _ =>
    _.then(L.castArray)
      .then(asTable)
      .then(log)

  return {
    t,
    t1,
    js,
    o
  }
}
export default prelude
