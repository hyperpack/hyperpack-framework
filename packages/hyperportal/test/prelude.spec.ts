import { HyperModel } from '@hyperpack/hypermodel'
import { PrimaryGeneratedColumn, Column, Index, Entity } from 'typeorm'
import prelude from '../src/prelude'

class Band extends HyperModel {
  @Column()
  frontman: string
}
const passthru = prelude(x => x)
const maiden = new Band()
maiden.frontman = 'bruce'

const o = Promise.resolve(maiden)

describe('prelude', () => {
  describe('t', () => {
    it('render', async () => {
      expect(await passthru.t(o)).toMatchSnapshot()
    })
  })
  describe('o', () => {
    it('render', async () => {
      expect(await passthru.o(o)).toMatchSnapshot()
    })
  })
  describe('t1', () => {
    it('render', async () => {
      expect(await passthru.t1(o)).toMatchSnapshot()
    })
  })
  describe('js', () => {
    it('render', async () => {
      expect(await passthru.js(o)).toMatchSnapshot()
    })
  })
})
