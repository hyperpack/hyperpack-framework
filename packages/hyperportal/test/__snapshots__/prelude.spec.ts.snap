// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prelude js render 1`] = `
"{
    [1m[37mfrontman[39m[22m: [33m'bruce'[39m
}"
`;

exports[`prelude o render 1`] = `
"{
    [1m[37mfrontman[39m[22m: [33m'bruce'[39m
}"
`;

exports[`prelude t render 1`] = `
"╔══════════╗
║ frontman ║
╟──────────╢
║ bruce    ║
╚══════════╝
"
`;

exports[`prelude t1 render 1`] = `
"frontman
--------
bruce   "
`;
